### PR TITLE
events: Add locks around publisher and subscriber name accesses

### PR DESCRIPTION
**Pull request by theopolis**
The sanitize-thread builds detect critical sections when running Python's `example_queries`. These additional locks protect the names and publisher/subscriber members.

### CODE REVIEW

**Comment by osqueryer** [comment]
:-1: The commit 9e6bb8b278b31d0adc7d2540c6468d8e39a37aea (Job results: [4118](https://jenkins.osquery.io/job/osqueryPullRequestLinux/4118/)) failed one or more tests (**Linux**).

File: osquery/events/events.cpp [hunk 1]
```diff
@@ -867,9 +876,9 @@ void EventFactory::end(bool join) {
   auto& ef = EventFactory::getInstance();
 
   {
```
Comment (theopolis) [inline osquery/events/events.cpp]: This scope can be removed.

**Comment by osqueryer** [comment]
:no_good: The commit 90521c6523a0cde9fb76f5d6bb6bc3073e798201 (Job results: [1062](https://jenkins.osquery.io/job/osqueryPullRequestAudit/1062/)) failed the code audit or documentation test.

**Comment by osqueryer** [comment]
:-1: The commit 90521c6523a0cde9fb76f5d6bb6bc3073e798201 (Job results: [361](https://jenkins.osquery.io/job/osqueryPullRequestOSX/361/)) failed one or more tests (**macOS/OS X**).

**Comment by osqueryer** [comment]
:-1: The commit 90521c6523a0cde9fb76f5d6bb6bc3073e798201 (Job results: [4124](https://jenkins.osquery.io/job/osqueryPullRequestLinux/4124/)) failed one or more tests (**Linux**).

**Comment by osqueryer** [comment]
:-1: The commit 90521c6523a0cde9fb76f5d6bb6bc3073e798201 (Job results: [1068](https://jenkins.osquery.io/job/osqueryPullRequestWindows/1068/)) failed one or more tests (**Windows**).

**Comment by theopolis** [comment]
ok to test

**Pull request merged**